src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final DtoResponseMapper<OrderResponseDto, Order> dtoResponseMapper;
+
+    @Autowired
+    public OrderController(
+            OrderService orderService, ShoppingCartService shoppingCartService,
+            UserService userService,
+            DtoResponseMapper<OrderResponseDto, Order> dtoResponseMapper
+    ) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.dtoResponseMapper = dtoResponseMapper;
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto compete(@RequestParam Long userId) {
+        return dtoResponseMapper.toDto(
+                orderService.completeOrder(shoppingCartService.getByUser(userService.get(userId))));
```

Comment:
I'd split this statement into several ones for more readability )
(Not only in this place)